src/python/pants/backend/python/rules/run_setup_py_test.py

```diff
@@ -56,6 +56,64 @@ def tgt(self, addr: str) -> HydratedTarget:
     return self.request_single_product(HydratedTarget, Params(Address.parse(addr)))
 
 
+class TestGetAncestorInitPy(TestSetupPyBase):
+  @classmethod
+  def rules(cls):
+    return super().rules() + [
+      get_ancestor_init_py,
+      RootRule(HydratedTargets),
+      RootRule(SourceRootConfig),
+    ]
+
+  def assert_ancestor_init_py(self, expected_init_pys, addrs):
```

Comment:
Type hints appreciated. Think this would be

```python
def assert_ancestor_init_py(self, expected_init_pys: Iterable[str], addrs: Iterable[str]) -> None:
```